ports/esp32s2/Makefile

```diff
@@ -0,0 +1,279 @@
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+
+# Select the board to build for.
+ifeq ($(BOARD),)
+  $(error You must provide a BOARD parameter)
+else
+  ifeq ($(wildcard boards/$(BOARD)/.),)
+    $(error Invalid BOARD specified)
+  endif
+endif
+
+# If the build directory is not given, make it reflect the board name.
+BUILD ?= build-$(BOARD)
+
+include ../../py/mkenv.mk
+# Board-specific
+include boards/$(BOARD)/mpconfigboard.mk
+# Port-specific
+include mpconfigport.mk
+
+# CircuitPython-specific
+include $(TOP)/py/circuitpy_mpconfig.mk
+
+# qstr definitions (must come before including py.mk)
+QSTR_DEFS = qstrdefsport.h
+
+# include py core make definitions
+include $(TOP)/py/py.mk
+
+include $(TOP)/supervisor/supervisor.mk
+
+# Include make rules and variables common across CircuitPython builds.
+include $(TOP)/py/circuitpy_defns.mk
+
+CROSS_COMPILE = xtensa-esp32s2-elf-
+
+#######################################
+# CFLAGS
+#######################################
+
+INC += -I.
+INC += -I../..
+INC += -I$(BUILD)
+INC += -I$(BUILD)/genhdr
+INC += -I./boards
+INC += -I./boards/$(BOARD)
+INC += -I./peripherals
+INC += -I../../lib/mp-readline
+INC += -I../../lib/tinyusb/src
+INC += -I../../supervisor/shared/usb
+INC += -Iesp-idf/components/freertos/include/freertos
+INC += -Iesp-idf/components/freertos/xtensa/include
+INC += -Iesp-idf/components/esp32s2/include
+INC += -Iesp-idf/components/xtensa/esp32s2/include
+INC += -Iesp-idf/components/esp_common/include
+INC += -Iesp-idf/components/esp_rom/include
+INC += -Iesp-idf/components/xtensa/include
+INC += -Iesp-idf/components/esp_timer/include
+INC += -Iesp-idf/components/soc/include
+INC += -Iesp-idf/components/soc/src/esp32s2/include
+INC += -Iesp-idf/components/soc/soc/include
+INC += -Iesp-idf/components/soc/soc/esp32s2/include
+INC += -Iesp-idf/components/heap/include
+INC += -Iesp-idf/components/esp_system/include
+INC += -I$(BUILD)/esp-idf/config
+
+CFLAGS += -DHAVE_CONFIG_H \
+          -DMBEDTLS_CONFIG_FILE=\"mbedtls/esp_config.h\" \
+          -DUNITY_INCLUDE_CONFIG_H -DWITH_POSIX
+
+# Make our canary value match FreeRTOS's
```

Comment:
You probably dislike "magic numbers" in your code as anyone else, and it would be a challenge ( but not impossible ) to extract the actual value from
tasks.c  ( in esp-idf/components/freertos )

329:#define tskSTACK_FILL_BYTE  ( 0xa5U )

I would humbly suggest adding a mention of  tskSTACK_FILL_BYTE  in the comment below
 ( it would have been nicer if freertos exported their symbol or value in some .h or API call so an assert or compile time check could be use to 
associate  STACK_CANARY_VALUE with  tskSTACK_FILL_BYTE  

ow that I look at it - over in micropython, there is a .h file that uses the the value defined in the .c file - ( I guess that implies that the tasks.c is the only file that includes include/freertos/StackMacros.h, and yes - I just checked - that is the case -
  -- must stop Yak shaving :-) )

https://github.com/micropython/micropython/compare/3d5d273cfe1016397eadb87c24d61937291be2db...tannewt:esp32s2 

 
https://github.com/tannewt/circuitpython/blob/esp32s2/ports/esp32s2/Makefile

Reply:
Done!